crates/ark/src/treesitter.rs

```diff
@@ -386,6 +389,36 @@ impl NodeTypeExt for Node<'_> {
     fn is_binary_operator(&self) -> bool {
         matches!(self.node_type(), NodeType::BinaryOperator(_))
     }
+
+    fn is_native_pipe_operator(&self) -> bool {
+        self.node_type() == NodeType::BinaryOperator(BinaryOperatorType::Pipe)
+    }
+
+    fn is_magrittr_pipe_operator(&self, contents: &ropey::Rope) -> anyhow::Result<bool> {
```

Comment:
One thing I did differently was to propagate this `Result` upward, which affected some design decisions elsewhere. It seems like the caller should get to decide what to do here.